libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1488,7 +1488,7 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	Token op = _operation.getOperator();
 	bool const modifying = (op == Token::Inc || op == Token::Dec || op == Token::Delete);
 	if (modifying)
-		requireLValue(_operation.subExpression());
+		requireLValue(_operation.subExpression(), op == Token::Inc || op == Token::Dec);
```

Comment:
`delete` does not actually read, but it should also not serve as a regular assignment for immutables. Maybe we change the flag instead to denote "ordinary assignment or not"?